src/main/java/com/parasoft/demoapp/controller/LoginController.java

```diff
@@ -30,4 +33,16 @@ public String showLoginPage(Authentication authentication, ModelMap modelMap) {
 
         return "login";
     }
+
+    @GetMapping("/keycloak")
+    public String validateKeycloakServerUrl() throws IOException {
+        String keycloakServerUrl = "/oauth2/authorization/keycloak";
+        try {
+            UrlUtil.validateUrl("http://localhost:8080"+ keycloakServerUrl);
```

Comment:
Validate the value of `spring.security.oauth2.client.provider.keycloak.issuer-uri`.